hathor/consensus/poa/poa_block_producer.py

```diff
@@ -144,7 +154,8 @@ def _produce_block(self, previous_block: PoaBlock) -> None:
         block_templates = self.manager.get_block_templates(parent_block_hash=previous_block.hash)
         block = block_templates.generate_mining_block(self.manager.rng, cls=PoaBlock)
         assert isinstance(block, PoaBlock)
-        block.weight = poa.calculate_weight(self._poa_settings, block, self._signer_index)
+        signer_index = self._get_signer_index(previous_block)
+        block.weight = poa.calculate_weight(self._poa_settings, block, not_none(signer_index))
```

Comment:
You might want to create a weight that is inversely proportional to the index distance to prevent ties. So if two or more out of order signers propagate a block, the closer one would win.

Reply:
Done in a separate PR: https://github.com/HathorNetwork/hathor-core/pull/1080

Reply:
Ok! Thanks!